src/main/java/com/sdase/k8s/operator/mongodb/controller/MongoDbController.java

```diff
@@ -43,61 +46,72 @@ public MongoDbController(
   @Override
   public DeleteControl cleanup(
       MongoDbCustomResource resource, Context<MongoDbCustomResource> context) {
-    LOG.info(
-        "MongoDb {}/{} deleted",
-        resource.getMetadata().getNamespace(),
-        resource.getMetadata().getName());
-    try {
-      var deleteDatabaseTask = taskFactory.newDeleteTask(resource);
-      var userDropped =
-          mongoDbService.dropDatabaseUser(
-              deleteDatabaseTask.databaseName(), deleteDatabaseTask.username());
-      if (!userDropped) {
-        throw new IllegalStateException("Failed to drop user");
-      }
-      if (deleteDatabaseTask.pruneDb()) {
-        boolean databaseDeleted = mongoDbService.dropDatabase(deleteDatabaseTask.databaseName());
-        if (!databaseDeleted) {
-          if (context.getRetryInfo().map(RetryInfo::isLastAttempt).orElse(false)) {
-            LOG.warn(
-                "Last attempt to delete database {} failed. Skipping.",
-                deleteDatabaseTask.databaseName());
-            return DeleteControl.defaultDelete();
+    try (var ignoredMdcNamespace =
+            MDC.putCloseable(MDC_RESOURCE_NAMESPACE_KEY, resource.getMetadata().getNamespace());
+        var ignoredMdcName =
+            MDC.putCloseable(MDC_RESOURCE_NAME_KEY, resource.getMetadata().getName())) {
```

Comment:
`resource.getMetadata().getNamespace()` + `...getName()` is called multiple times in `cleanup` and `reconcile`. I'd rather extract the values as variables at the beginning of the methods.

Reply:
Changed … though I don't think, that it increases readability a lot. Even Sonar did not complain.